web-module/src/main/webapp/resources/structEditJS/dateTime/celements-date-time-field.js

```diff
@@ -292,203 +296,195 @@
-        const cssFiles = ['celRes/images/glyphicons-halflings/css/glyphicons-halflings.css'];
-        cssFiles.forEach(function(cssFile) {
-          const cssElem = new Element('link', {
-            'rel': 'stylesheet',
-            'media': 'all',
-            'type': 'text/css',
-            'href': '/file/resources/' + cssFile + '?version=' + versionTimeStamp
-          });
-          document.head.append(cssElem);
-        });
+        this.#addCssFiles(document.head, [
+          '/file/resources/celRes/images/glyphicons-halflings/css/glyphicons-halflings.css'
+        ]);
       }
 
-      _addCssFiles() {
-        const _me = this;
-        //HACK be sure to load the glyphicons-halflings.css in the html-page too.
-        //HACK Because font-face will not work in shadow dom otherwise.
-        const cssFiles = ['celRes/images/glyphicons-halflings/css/glyphicons-halflings.css',
-          'celJS/jquery%2Ddatetimepicker/jquery.datetimepicker.css'
-        ];
-        cssFiles.forEach(function(cssFile) {
-          const cssElem = new Element('link', {
+      #addCssFiles(elem, cssFiles) {
+        cssFiles.forEach((cssFile) => {
+          elem.appendChild(new Element('link', {
             'rel': 'stylesheet',
             'media': 'all',
             'type': 'text/css',
-            'href': '/file/resources/' + cssFile + '?version=' + versionTimeStamp
-          });
-          _me.shadowRoot.appendChild(cssElem);
-        });
-        const dateTimeCssElem = new Element('link', {
-          'rel': 'stylesheet',
-          'media': 'all',
-          'type': 'text/css',
-          'href': curScriptDir + 'celements-date-time-field.css' + '?version='
-            + versionTimeStamp
+            'href': cssFile + '?version=' + versionTimeStamp
+          }));
         });
-        _me.shadowRoot.appendChild(dateTimeCssElem);
       }
 
-      _addInputFields() {
-        const _me = this;
-        _me._datePart = new Element('input', {
-          'type': 'text',
-          'name': 'datePart',
-          'class': 'dateInputField',
-          'autocomplete': 'off'
-        });
-        _me.shadowRoot.appendChild(_me._datePart);
-        _me._timePart = new Element('input', {
+      #newDisplayInputElem(type) {
+        return new Element('input', {
           'type': 'text',
-          'name': 'timePart',
-          'class': 'timeInputField',
+          'name': type + 'Part',
+          'class': type + 'InputField',
           'autocomplete': 'off'
         });
-        _me.shadowRoot.appendChild(_me._timePart);
       }
 
-      _addPickerIcons() {
-        const _me = this;
-        _me._datePickerIcon = new Element('i', {
+      #addInputFields() {
+        this.shadowRoot.appendChild(this.datePart);
+        if (this.hasTime()) {
+          this.shadowRoot.appendChild(this.timePart);
+        }
+      }
+
+      #addPickerIcons() {
+        this.#datePickerIcon = this.#datePickerIcon ?? new Element('i', {

```

Comment:
refactoring this part please replace the prototypjs 'new Element' with document.createElement too?